src/transaction/components/TransactionProvider.tsx

```diff
@@ -157,6 +163,18 @@ export function TransactionProvider({
   const singleTransactionHash =
     writeContractTransactionHash || sendCallTransactionHash;
 
+  const capabilities = useMemo(() => {
+    if (isSponsored && paymaster) {
+      return {
+        paymasterService: { url: paymaster },
+        // this needs to be below so devs can override default paymaster
+        // with their personal paymaster in production playgroundd
+        ...transactionCapabilities,
+      };
+    }
+    return transactionCapabilities;
+  }, [isSponsored, paymaster, transactionCapabilities]);
```

Comment:
`paymaster` from `useOnchainKit` should be the CDP Paymaster by default, and a 3rd party if override is provided

we should just adjust the playground to use the `OnchainKitProvider` prop instead

Reply:
also does the spread operator override field values?

Reply:
ideally to prevent breaking changes we should override the global config paymaster with the `capabilities` prop (if passed in)

Reply:
transactionCapabilities will override if provided